### PR TITLE
Migrate project to yarn v3.

### PR DESCRIPTION
**Pull request by alexkim205**
## Changes

This PR adds a few configuration files that make our project compatible with the latest version of yarn (v3.1.1).

Yarn v3 introduces a [Zero-Install](https://yarnpkg.com/features/zero-installs) concept that commits caches of libraries to Github so that you don't need to run `yarn install` in the first place. There are several implications (mostly pros like faster build times, consistent dev environments, etc) this has, which are discussed [here](https://yarnpkg.com/features/pnp#fixing-node_modules) and [here](https://yarnpkg.com/features/zero-installs#concerns). 

The best way to review this PR is to filter out files with `.zip` extensions (compressed packages).

### Considerations

This PR should not break:

- [ ] Individual dev environments
- [ ] Deployment
- [ ] Github Actions CI Workflow

### CODE REVIEW

**Comment by alexkim205** [comment]
:/ This is harder than it looks. Ran into a lot of blockers so I think it's best to continue using yarn v1 for now. Learnings:

- Yarn zero-install complicates package management and makes it difficult to view/copy/modify files in `node_modules` package directories.
- Zipping and pushing zipped dependencies to Github adds unnecessary bloat to codebase and the time it takes to pull down code cancels out any performance advantages gained by zero-install.
- There isn't a lot of community support for yarn v3, lacking community plugins (i.e., GH actions), documentation, and troubleshooting resources.

tl;dr - Yarn v3 offers a lot of cool performance advantages but at what cost???

qed - Closing out this rabbithole and moving on 😰 . Leaving the branch available in case we potentially pursue something like this again.

**Comment by mariusandra** [comment]
That's a big rabbit hole.

I also had to delete the remote branch here, as otherwise just a `yarn pull` downloaded it in the background, and that's too much traffic.

image

**Comment by weyert** [comment]
Did you had a look at `pnpm` that's quite cool too.
